src/components/screens/preview/draft-preview-screen.tsx

```diff
@@ -0,0 +1,158 @@
+import { Metaform } from "generated/client";
+import React, { useContext, useState } from "react";
+import strings from "localization/strings";
+import { FieldValue } from "metaform-react/types";
+import DraftPreviewHeader from "../../preview/draft-preview-header";
+import DraftPreviewShareDialog from "../../preview/draft-preview-dialog";
+import { useParams } from "react-router-dom";
+import { ErrorContext } from "components/contexts/error-handler";
+import Api from "api";
+import { useApiClient } from "app/hooks";
+import MetaformUtils from "utils/metaform-utils";
+import GenericLoaderWrapper from "components/generic/generic-loader";
+import Form from "components/generic/form";
+import { Dictionary } from "@reduxjs/toolkit";
+import { Alert, Snackbar } from "@mui/material";
+
+/**
+ * Metaform editor preview component
+ */
+const DraftPreviewScreen: React.FC = () => {
+  const [ dialogOpen, setDialogOpen ] = useState(false);
+  const [ loading, setLoading ] = useState(false);
+  const [ draftForm, setDraftForm ] = useState<Metaform>(MetaformUtils.jsonToMetaform({}));
+  const [ formValues, setFormValues ] = useState<Dictionary<FieldValue>>({});
+  const [ linkCopied, setLinkCopied ] = useState(false);
+  const [ emailSent, setEmailSent ] = useState(false);
+
+  const apiClient = useApiClient(Api.getApiClient);
+  const { metaformsApi, versionsApi } = apiClient;
+  const { formSlug, draftId } = useParams();
+  const errorContext = useContext(ErrorContext);
+
+  const linkToShare = `${window.location.origin}/${formSlug}`;
+
+  /**
+   * Loads MetaformVersion to edit.
+   */
+  const loadMetaformVersion = async () => {
+    setLoading(true);
+
+    try {
+      const form = await metaformsApi.findMetaform({ metaformSlug: formSlug });
+      const draft = await versionsApi.findMetaformVersion({
+        metaformId: form.id!,
+        versionId: draftId!
+      });
+
+      const parsedDraft = draft.data as Metaform;
+      setDraftForm(parsedDraft);
+      // setFormValues(MetaformUtils.prepareFormValues(parsedDraft, formValues!, undefined));
+    } catch (e) {
+      errorContext.setError(strings.errorHandling.draftEditorScreen.findDraft, e);
+    }
+
+    setLoading(false);
+  };
```

Comment:
We should not load draft version from api, we should preview changes made to existing draft.

Reply:
Ps. Background color should be the same as in the public form screen. That way it would be authentic preview of the form in production.